apps/nsqd/nsqd.go

```diff
@@ -232,6 +238,10 @@ func (p *program) Start() error {
 	}
 	nsqd.Main()
 
+	// hook into addons
+	addons := contrib.NewEnabledNSQDAddons(opts.ModOpt, nsqd, nsqd.Logf)
```

Comment:
Ah, that makes sense ... nope, can't think of a better way at the moment.

These "contrib addons" are going to want to get information from the NSQD, and maybe some will want to make changes, e.g. to the list of nsqlookupd, or auth info for new connections. These various purposes will need some functions to be made public, which we'll consider on a case-by-case basis. An interface of these public functions which contrib modules can use sounds like it could be a pretty good idea.